### PR TITLE
refactor: Configuration files uniformly use Kebab style

### LINKED ISSUE #5261: Yaml中的配置文件统一使用SpringBoot官方推荐的Kebab风格
**你的特性请求和某个问题有关吗？请描述**

SpringBoot推荐配置参数使用 [Kebab风格](https://docs.spring.io/spring-boot/3.3-SNAPSHOT/reference/features/external-config.html#features.external-config.typesafe-configuration-properties.relaxed-binding)

目前Apollo的几个yaml中的配置是 camelCase+Kebab 混用，是否可以统一使用 Kebab 风格

**清晰简洁地描述一下你希望的解决方案**

**清晰简洁地描述一下这个特性的备选方案**

**其它背景**

在这里添加和这个特性请求有关的背景说明、截图

### PR DESCRIPTION
**Pull request by youngzil**
## What's the purpose of this PR

Optimize configuration property naming
- Changed camel case naming to hyphen naming to improve readability and consistency
- Updated configuration properties in multiple files, including Eureka, LDAP and other related configurations
- Modified the example configurations in Chinese and English documents to reflect the new naming method

## Which issue(s) this PR fixes:
Fixes #5261

## Brief changelog

XXXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

- [ ] Read the [Contributing Guide](https://github.com/apolloconfig/apollo/blob/master/CONTRIBUTING.md) before making this pull request.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Write necessary unit tests to verify the code.
- [ ] Run `mvn clean test` to make sure this pull request doesn't break anything.
- [ ] Update the [`CHANGES` log](https://github.com/apolloconfig/apollo/blob/master/CHANGES.md).

## Summary by CodeRabbit

- **New Features**
	- Introduced a new configuration file for OpenID Connect (OIDC) authentication.
	- Enhanced LDAP authentication with improved filtering capabilities.
	- Added global search functionality for administrators.

- **Documentation**
	- Updated user guides for Java SDK and distributed deployment with clearer instructions and new sections.
	- Standardized configuration key naming from camelCase to kebab-case for consistency across documents.
	- Expanded details on Apollo client configuration, including new environment requirements and integration methods.

- **Bug Fixes**
	- Corrected various configuration property names in sample YAML files to align with new standards.
	- Resolved issues related to duplicate comments and blank lines in configuration management.

### CODE REVIEW

File: apollo-configservice/src/main/resources/application.yml [hunk 1]
```diff
@@ -38,22 +38,22 @@ logging:
...
       # This setting will be overridden by eureka.service.url setting from ApolloConfigDB.ServerConfig or System Property
       # see com.ctrip.framework.apollo.biz.eureka.ApolloEurekaClientConfig
       defaultZone: http://${eureka.instance.hostname}:8080/eureka/
     healthcheck:
       enabled: true
-    eurekaServiceUrlPollIntervalSeconds: 60
+    eureka-service-url-poll-interval-seconds: 60
     fetch-registry: false
-    registerWithEureka: false
+    registe-with-eureka: false
```
Comment (nobodyiam) [inline apollo-configservice/src/main/resources/application.yml:56]: I think the major effort of this pull request is to verify whether the configurations still work after renaming. e.g. the key name is mis-spelled.

File: docs/zh/client/java-sdk-user-guide.md [hunk 2]
```diff
@@ -1051,13 +1051,13 @@ server.port = 8080
...
 eureka.client.healthcheck.enabled = true
-eureka.client.registerWithEureka = true
-eureka.client.fetchRegistry = true
-eureka.client.eurekaServiceUrlPollIntervalSeconds = 60
+eureka.client.register-with-eureka = true
+eureka.client.fetch-registry = true
+eureka.client.eureka-service-url-poll-interval-seconds = 60
 
-eureka.instance.preferIpAddress = true
+eureka.instance.prefer-ip-address = true
```
[the remark this answers is not in the document]
Comment (youngzil) [reply docs/zh/client/java-sdk-user-guide.md:1060]: This cannot be changed

File: docs/zh/deployment/distributed-deployment-guide.md [hunk 3]
```diff
@@ -1251,14 +1251,14 @@ config:
...
           urls:
           - "ldap://xxx.somedomain.com:389"
       ldap:
         mapping:
-          objectClass: "inetOrgPerson"
-          loginId: "uid"
-          userDisplayName: "cn"
+          object-class: "inetOrgPerson"
+          login-id: "uid"
+          user-display-name: "cn"
```
[the remark this answers is not in the document]
Comment (youngzil) [reply docs/zh/deployment/distributed-deployment-guide.md:1261]: I checked the indentation and found no problem

File: docs/en/client/java-sdk-user-guide.md [hunk 4]
```diff
@@ -1105,13 +1105,13 @@ server.port = 8080
...
 
-eureka.client.serviceUrl.defaultZone = http://127.0.0.1:8761/eureka/
+eureka.client.service-url.defaultZone = http://127.0.0.1:8761/eureka/
 eureka.client.healthcheck.enabled=true
-eureka.client.registerWithEureka = true
-eureka.client.fetchRegistry = true
-eureka.client.eurekaServiceUrlPollIntervalSeconds = 60
+eureka.client.register-with-eureka = true
+eureka.client.fetch-registry = true
+eureka.client.eureka-service-url-poll-interval-seconds = 60
```
[the remark this answers is not in the document]
Comment (youngzil) [reply docs/en/client/java-sdk-user-guide.md:1112]: This cannot be changed

File: apollo-configservice/src/main/resources/application.yml (hunk 1 above)
Comment (youngzil) [reply apollo-configservice/src/main/resources/application.yml]: Yes, I have tested and confirmed that the eureka and LDAP configuration is effective

**Review by nobodyiam — APPROVED** [review]
LGTM

**Pull request merged**